pkg/network/usm/ebpf_main.go

```diff
@@ -410,19 +353,41 @@ func (e *ebpfProgram) init(buf bytecode.AssetReader, options manager.Options) er
 			},
 		},
 	}
-	addBoolConst(&options, e.cfg.EnableHTTPMonitoring, "http_monitoring_enabled")
-	addBoolConst(&options, e.cfg.EnableHTTP2Monitoring, "http2_monitoring_enabled")
-	addBoolConst(&options, e.cfg.EnableKafkaMonitoring, "kafka_monitoring_enabled")
+
+	protocols.AddBoolConst(&options, e.cfg.EnableHTTPMonitoring, "http_monitoring_enabled")
```

Comment:
shouldn't it be scoped with http protocol creation?

Reply:
Ideally these const values initialisation would be with the protocol's creation, but they require the options struct to be created first, which is done differently depending on whether or not we are using CO-RE, RC, or Prebuilt, so they have to be set here. We are trying to find a better way to do this in the future.